packages/@pollyjs/adapter-node-http/src/-private/transport-wrapper.js

```diff
@@ -0,0 +1,283 @@
+  }
+
+  patch() {
+    // make sure it's not already patched
+    this.adapter.assert(
+      `The ${
+        this.name
+      } transport has already been patched, please stop any running Polly instances`,
+      !this.isPatched()
+    );
+
+    nativeRequestMapping.set(this.transport, this.transport.request);
+    this.transport.request = this.createRequestWrapper();
+  }
+
+  restore() {
+    this.adapter.assert(
+      `Cannot restore unpatched transport ${this.name}`,
+      this.isPatched()
+    );
+
+    this.transport.request = nativeRequestMapping.get(this.transport);
+    nativeRequestMapping.delete(this.transport);
+  }
+
+  getBodyFromChunks(chunks, headers) {
+    const { adapter } = this;
+
+    //  If we have headers and there is content-encoding it means that
+    //  the body shouldn't be merged but instead persisted as an array
+    //  of hex strings so that the responses can be mocked one by one.
+    if (headers && isContentEncoded(headers)) {
+      const hexChunks = chunks.map(chunk => {
+        if (!Buffer.isBuffer(chunk)) {
+          adapter.assert(
+            'content-encoded responses must all be binary buffers',
+            typeof chunk === 'string'
+          );
+          chunk = Buffer.from(chunk);
+        }
+
+        return chunk.toString('hex');
+      });
+
+      return JSON.stringify(hexChunks);
+    }
+
+    const buffer = mergeChunks(chunks);
+
+    // The merged buffer can be one of two things:
+    //   1.  A binary buffer which then has to be recorded as a hex string.
+    //   2.  A string buffer.
+    return buffer.toString(isBinaryBuffer(buffer) ? 'hex' : 'utf8');
+  }
+
+  getChunksFromBody(body, headers) {
+    if (!body) {
+      return [];
+    }
+
+    if (Buffer.isBuffer(body)) {
+      return [body];
+    }
+
+    //  If we have headers and there is content-encoding it means that
+    //  the body is as an array of hex strings
+    if (headers && isContentEncoded(headers)) {
+      const hexChunks = JSON.parse(body);
+
+      return hexChunks.map(chunk => Buffer.from(chunk, 'hex'));
+    }
+
+    const buffer = Buffer.from(body);
+
+    // The body can be one of two things:
+    //   1.  A hex string which then means its binary data.
+    //   2.  A utf8 string which means a regular string.
+    return [Buffer.from(buffer, isBinaryBuffer(buffer) ? 'hex' : 'utf8')];
+  }
+
+  async passthrough(pollyRequest) {
+    const { transport } = this;
+    const [, , args] = pollyRequest.requestArguments;
+    const { method, headers, body: chunks } = pollyRequest;
+    const nativeRequest = nativeRequestMapping.get(transport);
+    let [url, options] = args;
+
+    /**
+     * args could be (url, options, callback) or (options, callback) depending
+     * on nodejs version. If the `url` arguments is not a string or URL
+     * instance, then use (options, callback).
+     */
+    if (typeof url !== 'string' || !(isObjectLike(url) && url.href)) {
```

Comment:
I think there might be a problem here, if url is a string it is used as options:
![image](https://user-images.githubusercontent.com/1007479/48974774-f3aded00-f072-11e8-8b91-ec238401d901.png)
 
and in the next lines it will be spread to `nativeRequest` call:
```js
const request = nativeRequest.call(transport, {
  ...options,
  method,
  headers: { ...headers },
  ...NodeURL.parse(pollyRequest.url)
});
```

Reply:
Yeah it was 3am when I was working on that part lol. I'll fix it 😄